Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.cs

```diff
@@ -508,7 +508,8 @@ protected void MuzzleFlash(EntityUid gun, AmmoComponent component, Angle worldAn
             return;
 
         var ev = new MuzzleFlashEvent(GetNetEntity(gun), sprite, worldAngle);
-        CreateEffect(gun, ev, gun);
+        var sending = _netManager.IsServer && user != null ? user : gun;
+        CreateEffect(gun, ev, sending);
```

Comment:
This just be
```suggestion
        CreateEffect(gun, ev, user);
```
And assuming the client-side effect is always meant to track the gun, it should just use the gun argument and rename the `tracked` argument back to `user` so that it's consistent with the shared & server-side methods